### PR TITLE
Set HTTP Version 1.1 on Post/Redirect/Get (PRG)

### LINKED ISSUE #769: redirect()->to() problem with output buffering
I have an AdminController class that all the rest of the controllers will extend.
in the constructor of  AdminController I check if the user is already logged, otherwise...

`redirect()->to('/admin/user/login');`

but it doesn't get redirected, even though I see the 'Location;' header sent to the browser correctly.
Seems to be a problem with the output buffering

For it to work I should do

```
redirect()->to('/admin/user/login');
ob_end_flush();
```
should we make the redirect()->to() method call ob_end_flush()?

Thanks in advance!

### LINKED ISSUE #931: Redirects not working
Trying to use `redirect()->back();` after an POST request the HTTP Header are show on the page body, like:

```
HTTP/ 303 See Other
Host: localhost:8080
Date: Thu, 08 Feb 2018 09:56:58 -0200
Connection: close
X-Powered-By: PHP/7.2.2-1
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Cache-control: no-store, max-age=0, no-cache
Content-Type: text/html; charset=UTF-8
Location: http://localhost:8080/login
```

The motive is that the HTTP Version is not set.

Looking on `system/HTTP/Response.php`, line 597, the `sendHeaders()` is called. And adding `$this->setProtocolVersion('1.1');` just before could solve this issue.

But I'm not sure if is the correct place to set it.

### PR DESCRIPTION
**Pull request by natanfelles**
Fix #769
Fix #931

### CODE REVIEW

**Comment by natanfelles** [comment]
Inside a Controller the Validation Errors are not accessible in `$validator`. But is possible to get the last errors (*_ci_validation_errors* flashdata) with `\Config\Services::validation()->getErrors()`.

What about after `redirect()->back()->withInput()` use the shared instance of the Validation with the `$validator`? Then enable to get the errors with `$this->validator->getErrors()`.

---

This could be done in the Controller constructor:

```php
$this->validator = Services::validation();
```

That will load the Validation every call, maybe check if has *_ci_validation_errors* flashdata (what could say if is coming back).

**Comment by lonnieezell** [comment]
A couple of things: 

We shouldn't fix the HTTP version - 1.0 is theoretically possible to run into, and 2.0 is the new hotness. I'm pretty sure I was checking the incoming version at some point, but will have to check that later today when I'm not working the day job. We should be matching the incoming HTTP version or defaulting to 1.1.

The validator is a subject for a different thread, but the obvious problem being the values wouldn't survive the redirect. Additionally, I don't believe the validator is necessarily shared between models, and not 100% sure if that would be a good thing.

File: system/HTTP/Response.php [hunk 1]
```diff
@@ -573,6 +573,7 @@ public function redirect(string $uri, string $method = 'auto', int $code = null)
 		{
 			if (isset($_SERVER['SERVER_PROTOCOL'], $_SERVER['REQUEST_METHOD']) && $_SERVER['SERVER_PROTOCOL'] === 'HTTP/1.1')
```
Comment (natanfelles) [inline system/HTTP/Response.php]: The version 1.1 is checked here. Maybe remove the `HTTP/` and get the numbers to set below.

Comment (natanfelles) [reply system/HTTP/Response.php]: But the conditional forces to be 1.1...

**Comment by lonnieezell** [comment]
The thing to check first, I believe, is that the HTTP status is being sent correctly in the request, and that CI is copying that forward in to the response. This should happen [in CodeIgniter.php](https://github.com/bcit-ci/CodeIgniter4/blob/develop/system/CodeIgniter.php#L390). 

If that's being set correctly there, it should already exist at the time of the redirect. But, yes, it looks like it needs to get checked properly during the redirect code.

**Comment by natanfelles** [comment]
Thank you! 😄

**Pull request closed without merging**